v2/reader.go

```diff
@@ -0,0 +1,61 @@
+package car
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+
+	carv1 "github.com/ipld/go-car"
+)
+
+const version2 = 2
+
+// Reader represents a reader of CAR v2.
+type Reader struct {
+	Header Header
+	r      io.ReaderAt
+}
+
+// NewReader constructs a new reader that reads CAR v2 from the given r.
+// Upon instantiation, the reader inspects the payload by reading the first 11 bytes and will return
+// an error if the payload does not represent a CAR v2.
+func NewReader(r io.ReaderAt) (*Reader, error) {
+	cr := &Reader{
+		r: r,
+	}
+	if err := cr.readPrefix(); err != nil {
+		return nil, err
+	}
+	if err := cr.readHeader(); err != nil {
+		return nil, err
+	}
+	return cr, nil
+}
+
+func (r *Reader) readPrefix() (err error) {
```

Comment:
oo, this is a nice touch, nicely forward compatible and involves carv1 :ok_hand: